presenters/thread_utils.rb

```diff
@@ -17,6 +17,23 @@ def self.get_endorsed(threads)
     endorsed_threads
   end
 
+  def self.get_abuse_flagged_count(threads)
+    # returns a count of flagged threads for all comments within the thread
+    # only threads with at least one flag are returned
+    flagged_threads = {}
+    thread_ids = threads.collect {|t| t._id}
+    Comment.collection.aggregate([
+      {"$match" => {
+        "comment_thread_id" => {"$in" => thread_ids},
+        "abuse_flaggers"=> {"$ne"=> []},
+      }},
+      {"$group" => {"_id" => "$comment_thread_id", "flagged" => { "$sum" => 1}} }
+    ]).each do |res|
```

Comment:
Copying details from the othr PR.

The way this query works is as follows:
1. The aggregate method [expects a `pipeline`, or an array of operation](https://docs.mongodb.com/manual/reference/method/db.collection.aggregate/)
2. The first operation, is [filtering entries that match](https://docs.mongodb.com/manual/reference/operator/aggregation/match/) the specified `comment_thread_id` and where the `abuse_flaggers` list is [not equal](https://docs.mongodb.com/manual/reference/operator/query/ne/)  to an empty list, i.e. it has abuse flaggers. 
3. The second operation is a [grouping operation](https://docs.mongodb.com/manual/reference/operator/aggregation/group/), where the results are grouped by the comment id. For each group we're also adding up the number of thread with [$sum](https://docs.mongodb.com/manual/reference/operator/aggregation/sum/#mongodb-group-grp.-sum). 

The result is a list that has thread ids and the number of flagged comments in them. This is then converted to a dict/map/hash of the thread id to the count in the next line.